Flaky test TestCassandraConnectorTest.testSelectInformationSchemaColumns, testSelectInformationSchemaTables

```
Error:  io.trino.plugin.cassandra.TestCassandraConnectorTest.testSelectInformationSchemaColumns -- Time elapsed: 3.041 s <<< FAILURE!
java.lang.AssertionError: Expected query 20240325_213705_00350_hc5v2 to succeed: SELECT * FROM information_schema.columns
	at io.trino.testing.QueryAssertions.assertQuerySucceeds(QueryAssertions.java:450)
	at io.trino.testing.AbstractTestQueryFramework.assertQuerySucceeds(AbstractTestQueryFramework.java:435)
	at io.trino.testing.AbstractTestQueryFramework.assertQuerySucceeds(AbstractTestQueryFramework.java:430)
	at io.trino.testing.BaseConnectorTest.testSelectInformationSchemaColumns(BaseConnectorTest.java:2116)
	at java.base/java.lang.reflect.Method.invoke(Method.java:580)
	at java.base/java.util.concurrent.RecursiveAction.exec(RecursiveAction.java:194)
	at java.base/java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:387)
	at java.base/java.util.concurrent.ForkJoinPool.helpJoin(ForkJoinPool.java:2076)
	at java.base/java.util.concurrent.ForkJoinTask.awaitDone(ForkJoinTask.java:423)
	at java.base/java.util.concurrent.ForkJoinTask.join(ForkJoinTask.java:651)
	at java.base/java.util.concurrent.RecursiveAction.exec(RecursiveAction.java:194)
	at java.base/java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:387)
	at java.base/java.util.concurrent.ForkJoinPool$WorkQueue.tryRemoveAndExec(ForkJoinPool.java:1351)
	at java.base/java.util.concurrent.ForkJoinTask.awaitDone(ForkJoinTask.java:422)
	at java.base/java.util.concurrent.ForkJoinTask.join(ForkJoinTask.java:651)
	at java.base/java.util.concurrent.RecursiveAction.exec(RecursiveAction.java:194)
	at java.base/java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:387)
	at java.base/java.util.concurrent.ForkJoinPool$WorkQueue.topLevelExec(ForkJoinPool.java:1312)
	at java.base/java.util.concurrent.ForkJoinPool.scan(ForkJoinPool.java:1843)
	at java.base/java.util.concurrent.ForkJoinPool.runWorker(ForkJoinPool.java:1808)
	at java.base/java.util.concurrent.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:188)
Caused by: io.trino.testing.QueryFailedException: Error listing table columns for catalog cassandra: More than one keyspace has been found for the case insensitive schema name: keyspace_3 -> (KeYsPaCe_3, kEySpAcE_3)
	at io.trino.testing.AbstractTestingTrinoClient.execute(AbstractTestingTrinoClient.java:133)
	at io.trino.testing.DistributedQueryRunner.executeInternal(DistributedQueryRunner.java:508)
	at io.trino.testing.DistributedQueryRunner.execute(DistributedQueryRunner.java:491)
	at io.trino.testing.QueryAssertions.assertQuerySucceeds(QueryAssertions.java:447)
	... 20 more
	Suppressed: java.lang.Exception: SQL: SELECT * FROM information_schema.columns
		at io.trino.testing.DistributedQueryRunner.executeInternal(DistributedQueryRunner.java:515)
		... 22 more
Caused by: io.trino.spi.TrinoException: Error listing table columns for catalog cassandra: More than one keyspace has been found for the case insensitive schema name: keyspace_3 -> (KeYsPaCe_3, kEySpAcE_3)
	at io.trino.metadata.MetadataListing.handleListingException(MetadataListing.java:358)
	at io.trino.metadata.MetadataListing.listTableColumns(MetadataListing.java:254)
	at io.trino.connector.informationschema.InformationSchemaPageSource.addColumnsRecords(InformationSchemaPageSource.java:250)
	at io.trino.connector.informationschema.InformationSchemaPageSource.buildPages(InformationSchemaPageSource.java:218)
	at io.trino.connector.informationschema.InformationSchemaPageSource.getNextPage(InformationSchemaPageSource.java:185)
	at io.trino.operator.TableScanOperator.getOutput(TableScanOperator.java:291)
	at io.trino.operator.Driver.processInternal(Driver.java:395)
	at io.trino.operator.Driver.lambda$process$8(Driver.java:298)
	at io.trino.operator.Driver.tryWithLock(Driver.java:701)
	at io.trino.operator.Driver.process(Driver.java:290)
	at io.trino.operator.Driver.processForDuration(Driver.java:261)
	at io.trino.execution.SqlTaskExecution$DriverSplitRunner.processFor(SqlTaskExecution.java:887)
	at io.trino.execution.executor.timesharing.PrioritizedSplitRunner.process(PrioritizedSplitRunner.java:187)
	at io.trino.execution.executor.timesharing.TimeSharingTaskExecutor$TaskRunner.run(TimeSharingTaskExecutor.java:565)
	at io.trino.$gen.Trino_testversion____20240325_213223_186.run(Unknown Source)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1144)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:642)
	at java.base/java.lang.Thread.run(Thread.java:1583)
Caused by: io.trino.spi.TrinoException: More than one keyspace has been found for the case insensitive schema name: keyspace_3 -> (KeYsPaCe_3, kEySpAcE_3)
	at io.trino.plugin.cassandra.CassandraSession.getKeyspaceByCaseInsensitiveName(CassandraSession.java:292)
	at io.trino.plugin.cassandra.CassandraSession.getCaseSensitiveTableNames(CassandraSession.java:193)
	at io.trino.plugin.cassandra.CassandraMetadata.listTables(CassandraMetadata.java:169)
	at io.trino.plugin.cassandra.CassandraMetadata.listTables(CassandraMetadata.java:218)
	at io.trino.plugin.cassandra.CassandraMetadata.listTableColumns(CassandraMetadata.java:204)
	at io.trino.spi.connector.ConnectorMetadata.streamTableColumns(ConnectorMetadata.java:349)
	at io.trino.spi.connector.ConnectorMetadata.streamRelationColumns(ConnectorMetadata.java:370)
	at io.trino.tracing.TracingConnectorMetadata.streamRelationColumns(TracingConnectorMetadata.java:321)
	at io.trino.metadata.MetadataManager.listTableColumns(MetadataManager.java:669)
	at io.trino.tracing.TracingMetadata.listTableColumns(TracingMetadata.java:355)
	at io.trino.metadata.MetadataListing.doListTableColumns(MetadataListing.java:261)
	at io.trino.metadata.MetadataListing.listTableColumns(MetadataListing.java:251)
	... 16 more
```

```
Error:  io.trino.plugin.cassandra.TestCassandraConnectorTest.testSelectInformationSchemaColumns -- Time elapsed: 6.792 s <<< FAILURE!
java.lang.AssertionError: Execution of 'actual' query 20240411_213112_00270_z7dz3 failed: SELECT table_name, column_name FROM information_schema.columns WHERE table_schema = 'tpch' AND table_name LIKE '%rders'
	at io.trino.testing.QueryAssertions.assertDistributedQuery(QueryAssertions.java:299)
	at io.trino.testing.QueryAssertions.assertQuery(QueryAssertions.java:187)
	at io.trino.testing.QueryAssertions.assertQuery(QueryAssertions.java:160)
	at io.trino.testing.AbstractTestQueryFramework.assertQuery(AbstractTestQueryFramework.java:350)
	at io.trino.testing.BaseConnectorTest.testSelectInformationSchemaColumns(BaseConnectorTest.java:2081)
	at java.base/java.lang.reflect.Method.invoke(Method.java:580)
	at java.base/java.util.concurrent.RecursiveAction.exec(RecursiveAction.java:194)
	at java.base/java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:507)
	at java.base/java.util.concurrent.ForkJoinPool$WorkQueue.topLevelExec(ForkJoinPool.java:1491)
	at java.base/java.util.concurrent.ForkJoinPool.scan(ForkJoinPool.java:2073)
	at java.base/java.util.concurrent.ForkJoinPool.runWorker(ForkJoinPool.java:2035)
	at java.base/java.util.concurrent.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:187)
Caused by: io.trino.testing.QueryFailedException: Error listing table columns for catalog cassandra: More than one column has been found for the case insensitive column name: col_case -> (col_case, COL_CASE)
	at io.trino.testing.AbstractTestingTrinoClient.execute(AbstractTestingTrinoClient.java:133)
	at io.trino.testing.DistributedQueryRunner.executeInternal(DistributedQueryRunner.java:501)
	at io.trino.testing.DistributedQueryRunner.executeWithPlan(DistributedQueryRunner.java:490)
	at io.trino.testing.QueryAssertions.assertDistributedQuery(QueryAssertions.java:290)
	... 11 more
	Suppressed: java.lang.Exception: SQL: SELECT table_name, column_name FROM information_schema.columns WHERE table_schema = 'tpch' AND table_name LIKE '%rders'
		at io.trino.testing.DistributedQueryRunner.executeInternal(DistributedQueryRunner.java:508)
		... 13 more
Caused by: io.trino.spi.TrinoException: Error listing table columns for catalog cassandra: More than one column has been found for the case insensitive column name: col_case -> (col_case, COL_CASE)
	at io.trino.metadata.MetadataListing.handleListingException(MetadataListing.java:358)
	at io.trino.metadata.MetadataListing.listTableColumns(MetadataListing.java:254)
	at io.trino.connector.informationschema.InformationSchemaPageSource.addColumnsRecords(InformationSchemaPageSource.java:250)
	at io.trino.connector.informationschema.InformationSchemaPageSource.buildPages(InformationSchemaPageSource.java:218)
	at io.trino.connector.informationschema.InformationSchemaPageSource.getNextPage(InformationSchemaPageSource.java:185)
	at io.trino.operator.ScanFilterAndProjectOperator$ConnectorPageSourceToPages.process(ScanFilterAndProjectOperator.java:379)
	at io.trino.operator.WorkProcessorUtils$ProcessWorkProcessor.process(WorkProcessorUtils.java:412)
	at io.trino.operator.WorkProcessorUtils.getNextState(WorkProcessorUtils.java:261)
	at io.trino.operator.WorkProcessorUtils$YieldingProcess.process(WorkProcessorUtils.java:181)
	at io.trino.operator.WorkProcessorUtils$ProcessWorkProcessor.process(WorkProcessorUtils.java:412)
	at io.trino.operator.WorkProcessorUtils$3.process(WorkProcessorUtils.java:346)
	at io.trino.operator.WorkProcessorUtils$ProcessWorkProcessor.process(WorkProcessorUtils.java:412)
	at io.trino.operator.WorkProcessorUtils$3.process(WorkProcessorUtils.java:346)
	at io.trino.operator.WorkProcessorUtils$ProcessWorkProcessor.process(WorkProcessorUtils.java:412)
	at io.trino.operator.WorkProcessorUtils$3.process(WorkProcessorUtils.java:346)
	at io.trino.operator.WorkProcessorUtils$ProcessWorkProcessor.process(WorkProcessorUtils.java:412)
	at io.trino.operator.WorkProcessorUtils.getNextState(WorkProcessorUtils.java:261)
	at io.trino.operator.WorkProcessorUtils$BlockingProcess.process(WorkProcessorUtils.java:207)
	at io.trino.operator.WorkProcessorUtils$ProcessWorkProcessor.process(WorkProcessorUtils.java:412)
	at io.trino.operator.WorkProcessorUtils.lambda$flatten$6(WorkProcessorUtils.java:317)
	at io.trino.operator.WorkProcessorUtils$3.process(WorkProcessorUtils.java:359)
	at io.trino.operator.WorkProcessorUtils$ProcessWorkProcessor.process(WorkProcessorUtils.java:412)
	at io.trino.operator.WorkProcessorUtils$3.process(WorkProcessorUtils.java:346)
	at io.trino.operator.WorkProcessorUtils$ProcessWorkProcessor.process(WorkProcessorUtils.java:412)
	at io.trino.operator.WorkProcessorUtils.getNextState(WorkProcessorUtils.java:261)
	at io.trino.operator.WorkProcessorUtils.lambda$processStateMonitor$2(WorkProcessorUtils.java:240)
	at io.trino.operator.WorkProcessorUtils$ProcessWorkProcessor.process(WorkProcessorUtils.java:412)
	at io.trino.operator.WorkProcessorUtils.getNextState(WorkProcessorUtils.java:261)
	at io.trino.operator.WorkProcessorUtils.lambda$finishWhen$3(WorkProcessorUtils.java:255)
	at io.trino.operator.WorkProcessorUtils$ProcessWorkProcessor.process(WorkProcessorUtils.java:412)
	at io.trino.operator.WorkProcessorSourceOperatorAdapter.getOutput(WorkProcessorSourceOperatorAdapter.java:133)
	at io.trino.operator.Driver.processInternal(Driver.java:403)
	at io.trino.operator.Driver.lambda$process$8(Driver.java:306)
	at io.trino.operator.Driver.tryWithLock(Driver.java:709)
	at io.trino.operator.Driver.process(Driver.java:298)
	at io.trino.operator.Driver.processForDuration(Driver.java:269)
	at io.trino.execution.SqlTaskExecution$DriverSplitRunner.processFor(SqlTaskExecution.java:890)
	at io.trino.execution.executor.dedicated.SplitProcessor.run(SplitProcessor.java:76)
	at io.trino.execution.executor.dedicated.TaskEntry$VersionEmbedderBridge.lambda$run$0(TaskEntry.java:191)
	at io.trino.$gen.Trino_testversion____20240411_21[293](https://github.com/trinodb/trino/actions/runs/8652650014/job/23727019603?pr=21522#step:5:294)3_151.run(Unknown Source)
	at io.trino.execution.executor.dedicated.TaskEntry$VersionEmbedderBridge.run(TaskEntry.java:192)
	at io.trino.execution.executor.scheduler.FairScheduler.runTask(FairScheduler.java:174)
	at io.trino.execution.executor.scheduler.FairScheduler.lambda$submit$0(FairScheduler.java:161)
	at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:572)
	at com.google.common.util.concurrent.TrustedListenableFutureTask$TrustedFutureInterruptibleTask.runInterruptibly(TrustedListenableFutureTask.java:131)
	at com.google.common.util.concurrent.InterruptibleTask.run(InterruptibleTask.java:76)
	at com.google.common.util.concurrent.TrustedListenableFutureTask.run(TrustedListenableFutureTask.java:82)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1144)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:642)
	at java.base/java.lang.Thread.run(Thread.java:1570)
Caused by: io.trino.spi.TrinoException: More than one column has been found for the case insensitive column name: col_case -> (col_case, COL_CASE)
	at io.trino.plugin.cassandra.CassandraSession.checkColumnNames(CassandraSession.java:341)
	at io.trino.plugin.cassandra.CassandraSession.getTable(CassandraSession.java:212)
	at io.trino.plugin.cassandra.CassandraMetadata.getTableMetadata(CassandraMetadata.java:155)
	at io.trino.plugin.cassandra.CassandraMetadata.listTableColumns(CassandraMetadata.java:206)
	at io.trino.spi.connector.ConnectorMetadata.streamTableColumns(ConnectorMetadata.java:348)
	at io.trino.spi.connector.ConnectorMetadata.streamRelationColumns(ConnectorMetadata.java:369)
	at io.trino.tracing.TracingConnectorMetadata.streamRelationColumns(TracingConnectorMetadata.java:321)
	at io.trino.metadata.MetadataManager.listTableColumns(MetadataManager.java:669)
	at io.trino.tracing.TracingMetadata.listTableColumns(TracingMetadata.java:355)
	at io.trino.metadata.MetadataListing.doListTableColumns(MetadataListing.java:261)
	at io.trino.metadata.MetadataListing.listTableColumns(MetadataListing.java:251)
	... 48 more
```

--- a/plugin/trino-cassandra/src/test/java/io/trino/plugin/cassandra/TestCassandraConnectorTest.java
+++ b/plugin/trino-cassandra/src/test/java/io/trino/plugin/cassandra/TestCassandraConnectorTest.java
@@ -932,28 +932,25 @@ public class TestCassandraConnectorTest
     @Test
     public void testKeyspaceNameAmbiguity()
     {
-        // Run exclusively as the test temporarily creates unsupported state (name conflict)
-        executeExclusively(() -> {
-            // Identifiers enclosed in double quotes are stored in Cassandra verbatim. It is possible to create 2 keyspaces with names
-            // that have differences only in letters case.
-            session.execute("CREATE KEYSPACE \"KeYsPaCe_3\" WITH REPLICATION = {'class':'SimpleStrategy', 'replication_factor': 1}");
-            session.execute("CREATE KEYSPACE \"kEySpAcE_3\" WITH REPLICATION = {'class':'SimpleStrategy', 'replication_factor': 1}");
+        // Identifiers enclosed in double quotes are stored in Cassandra verbatim. It is possible to create 2 keyspaces with names
+        // that have differences only in letters case.
+        session.execute("CREATE KEYSPACE \"KeYsPaCe_3\" WITH REPLICATION = {'class':'SimpleStrategy', 'replication_factor': 1}");
+        session.execute("CREATE KEYSPACE \"kEySpAcE_3\" WITH REPLICATION = {'class':'SimpleStrategy', 'replication_factor': 1}");
 
-            // Although in Trino all the schema and table names are always displayed as lowercase
-            assertContainsEventually(() -> execute("SHOW SCHEMAS FROM cassandra"), resultBuilder(getSession(), createUnboundedVarcharType())
-                    .row("keyspace_3")
-                    .row("keyspace_3")
-                    .build(), new Duration(1, MINUTES));
+        // Although in Trino all the schema and table names are always displayed as lowercase
+        assertContainsEventually(() -> execute("SHOW SCHEMAS FROM cassandra"), resultBuilder(getSession(), createUnboundedVarcharType())
+                .row("keyspace_3")
+                .row("keyspace_3")
+                .build(), new Duration(1, MINUTES));
 
-            // There is no way to figure out what the exactly keyspace we want to retrieve tables from
-            assertQueryFailsEventually(
-                    "SHOW TABLES FROM cassandra.keyspace_3",
-                    "Error listing tables for catalog cassandra: More than one keyspace has been found for the case insensitive schema name: keyspace_3 -> \\(KeYsPaCe_3, kEySpAcE_3\\)",
-                    new Duration(1, MINUTES));
+        // There is no way to figure out what the exactly keyspace we want to retrieve tables from
+        assertQueryFailsEventually(
+                "SHOW TABLES FROM cassandra.keyspace_3",
+                "Error listing tables for catalog cassandra: More than one keyspace has been found for the case insensitive schema name: keyspace_3 -> \\(KeYsPaCe_3, kEySpAcE_3\\)",
+                new Duration(1, MINUTES));
 
-            session.execute("DROP KEYSPACE \"KeYsPaCe_3\"");
-            session.execute("DROP KEYSPACE \"kEySpAcE_3\"");
-        });
+        session.execute("DROP KEYSPACE \"KeYsPaCe_3\"");
+        session.execute("DROP KEYSPACE \"kEySpAcE_3\"");
     }
 
     @Test

--- a/plugin/trino-cassandra/src/test/java/io/trino/plugin/cassandra/TestCassandraConnectorTest.java
+++ b/plugin/trino-cassandra/src/test/java/io/trino/plugin/cassandra/TestCassandraConnectorTest.java
@@ -223,6 +223,13 @@ public class TestCassandraConnectorTest
     }
 
     @Test
+    @Override
+    public void testSelectInformationSchemaColumns()
+    {
+        executeExclusively(super::testSelectInformationSchemaColumns);
+    }
+
+    @Test
     public void testPushdownUuidPartitionKeyPredicate()
     {
         try (TestCassandraTable testCassandraTable = testTable(

